packages/@markuplint/ml-core/src/ml-rule/types.ts

```diff
@@ -7,6 +7,8 @@ export interface MLRuleOptions<T extends RuleConfigValue, O = null> {
 	defaultLevel?: Severity;
 	defaultValue: T;
 	defaultOptions: O;
-	verify(document: Document<T, O>, translate: Translator, globalRule: RuleInfo<T, O>): Promise<Result[]>;
-	fix?(document: Document<T, O>, globalRule: RuleInfo<T, O>): Promise<void>;
+	verify(document: Document<T, O>, translate: Translator, globalRule: RuleInfo<T, O>): Promise<Result[]> | Result[];
+	verifySync(document: Document<T, O>, translate: Translator, globalRule: RuleInfo<T, O>): Result[];
```

Comment:
I don't know `verifySync` should be `required` or `optional` here, if `required`, the rule author is responsible to throw an error if it does not support `sync` API like `rule-textlint` for now; if `optional` we should detect `sync` API calls and warning if the rule only support `async` API like [unified](https://github.com/unifiedjs/unified).

Reply:
Or, we can mark `verify` as `optional` too, and reuse the `verifySync` method as fallback, but we should warn on the rule which doesn't provide `verify` and `verifySync` at the same time.

That is true to be because as I noticed that all rules are implemented in synchronized way, although they're marked as `async`.